src/bittybuzz/src/BittyBuzzClosureRegister.cpp

```diff
@@ -3,18 +3,21 @@
 #include <cstring>
 #include <string_view>
 
-BittyBuzzRegisteredClosure::BittyBuzzRegisteredClosure(const char* functionName,
+BittyBuzzRegisteredClosure::BittyBuzzRegisteredClosure(uint16_t closureId,
+                                                       BittyBuzzFunctionDescription description,
                                                        bbzheap_idx_t closureHeapIdx,
                                                        bbzheap_idx_t selfHeapIdx) :
-    m_closureHeapIdx(closureHeapIdx), m_selfHeapIdx(selfHeapIdx), m_description(functionName) {}
 
-BittyBuzzRegisteredClosure::BittyBuzzRegisteredClosure() : m_description("") {}
+    m_closureId(closureId),
+    m_closureHeapIdx(closureHeapIdx),
+    m_selfHeapIdx(selfHeapIdx),
+    m_description(std::move(description)) {}
 
 bool BittyBuzzClosureRegister::registerClosure(const char* functionName,
                                                bbzheap_idx_t closureHeapIdx,
                                                bbzheap_idx_t selfHeapIdx,
                                                const BittyBuzzFunctionDescription& description) {
-    if (m_closureRegistersLength >= m_maxSize) {
+    if (m_closureNameRegisters.isFull() || m_closureRegisterMap.isFull()) {
```

Comment:
To we want to keep older values or insert newer ones?

Reply:
Removed so we can overwrite existing even if the hashmap is full